clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/cache/Keys.groovy

```diff
@@ -56,6 +41,17 @@ class Keys {
               result << [account: parts[2], region: parts[3], instanceId: parts[4]]
             }
             break
+          case Namespace.APPLICATIONS.ns:
+            if (parts.length == 3) {
+              result << [application: parts[2].toLowerCase()]
+            }
+            break
+          case Namespace.CLUSTERS.ns:
+            if (parts.length == 5) {
+              def names = Names.parseName(parts[4])
+              result << [application: parts[3].toLowerCase(), account: parts[2], cluster: parts[4], stack: names.stack, detail: names.detail]
```

Comment:
I saw this and thought it would be nice for ensuring a caching key, but at the cost of lower casing everything.  I saw this on another implementation, is there a preferred approach?